app/src/main/java/de/tum/in/tumcampusapp/api/app/TUMCabeClient.java

```diff
@@ -365,65 +370,53 @@ public Event getEvent(int eventID) throws IOException {
-        return service.cancelTicketReservation(ticketHistory).execute().body();
+    public void cancelTicketReservation(Context context, int ticketHistory, Callback<TicketSuccessResponse> cb) throws IOException {
+        ChatVerification chatVerification = createChatVerification(context, new TicketReservationCancelation(ticketHistory));
+        service.cancelTicketReservation(chatVerification).enqueue(cb);
     }
 
     // Ticket purchase
-    public Ticket purchaseTicketStripe(int ticketHistory, String token, String customerMail,
-                                       String customerName) throws IOException {
-        return new Ticket(1, 2, "codeblablabla918gr182gr9128g2u8f393u2vu32",
-                1, false);
-        /*
+    public void purchaseTicketStripe(Context context, int ticketHistory, String token, String customerMail,
+                                       String customerName, Callback<Ticket> cb) throws IOException {
         HashMap<String, Object> argsMap = new HashMap<>();
         argsMap.put("ticket_history", ticketHistory);
         argsMap.put("token", token);
         argsMap.put("customer_mail", customerMail);
         argsMap.put("customer_name", customerName);
 
-        HashMap<String, Object> map = service.purchaseTicketStripe(argsMap).execute().body();
-        List<TicketType> ticketTypes = service.getTicketTypes((int) map.get("event")).execute().body();
-
-        // Directly create the associated ticket type as an object
-        TicketType tt = null;
-        for (TicketType curr : ticketTypes) {
-            if (curr.getId() == (int) map.get("ticket_type")) {
-                tt = curr;
-                break;
-            }
-        }
-        if (tt == null) {
-            throw new IOException();
-        }
+        ChatVerification chatVerification = createChatVerification(context, argsMap);
+        service.purchaseTicketStripe(chatVerification).enqueue(cb);
+    }
 
-        return new Ticket(EventsController.getEventById((int) map.get("event")),
-                (String) map.get("code"),
-                tt,
-                (int) map.get("ticket_payment"));
-                */
+    public void retrieveEphemeralKey(Context context, String apiVersion, String customerMail, Callback<HashMap<String, Object>> cb) throws IOException {
+        ChatVerification chatVerification = createChatVerification(context, new EphimeralKey(customerMail, apiVersion));
+        service.retrieveEphemeralKey(chatVerification).enqueue(cb);
     }
 
-    public HashMap<String, Object> retrieveEphemeralKey(String apiVersion, String customerMail) throws IOException {
-        HashMap<String, Object> map = new HashMap<>();
-        map.put("api_version", apiVersion);
-        map.put("customer_mail", customerMail);
-        return service.retrieveEphemeralKey(map).execute().body();
+    private ChatVerification createChatVerification(Context context, Object object) throws IOException {
```

Comment:
What happened here? This should not be the function name?! Or is the diff broken?

Reply:
@kordianbruck No it's not broken, this should be a helper function to create a ChatVerification object, since we need that for every endpoint that has authentication.

Reply:
Please put this into ChatVerofication then. Thats not the right place there.

Reply:
right, put it there.